local-tests/test_catch_up.py

```diff
@@ -36,29 +36,40 @@
 print('Waiting 90s')
 sleep(90)
 
-check_finalized(chain)
+finalized_before_kill_per_node = check_finalized(chain)
 print('Killing one validator and one nonvalidator')
+
 chain[3].stop()
 chain[4].stop()
 
-print('waiting around 4 sessions')
-sleep(30 * 4)
+print('waiting around 2 sessions')
+sleep(30 * 2)
 
 print('restarting nodes')
+finalized_before_start_per_node = check_finalized(chain)
+
+# Check if the finalization didnt stopped after kill
+if finalized_before_start_per_node[0] - finalized_before_kill_per_node[0] < 10:
+    print('Finalization stalled')
+    sys.exit(1)
+
 chain.start('aleph', nodes=[3, 4])
-check_finalized(chain)
+finalized_after_start = check_finalized(chain)
+
 print('waiting 45s for catch up')
 sleep(45)
-finalized_per_node = check_finalized(chain)
+finalized_after_catch_up_per_node = check_finalized(chain)
+
+nonvalidator_diff = finalized_after_catch_up_per_node[4] - finalized_before_start_per_node[0]
```

Comment:
Is there a reason to compare that between different nodes? Why not
`finalized_after_catch_up_per_node[4] - finalized_before_start_per_node[0]`?
Same applies to the line below

Reply:
did you meant `finalized_after_catch_up_per_node[4] - finalized_before_start_per_node[4]`?

Reply:
done